### ISSUE TITLE
-prune broken?

### ISSUE THREAD

**Issue by cannedprimates** [dashpay/dash#1466]
This error triggers when running with -prune=555 -txindex=0 (required by -prune):

https://github.com/dashpay/dash/blob/master/src/init.cpp#L1794

since masternode should default to false, I think that's a bug.

cheers!

**Comment by PanderMusubi** [comment]
Same happens with in `~/.dashcore/dash.conf` the line

    prune=555

for Dash Core version v0.12.1.5 (64-bit)

**Comment by PanderMusubi** [comment]
This error occurs only when -prune is given on the command line and/or when prune is in the config file.

**Comment by minerric** [comment]
I have same issue

i have prune=945 on dash.conf and txindex=0 and still get this error on Linux 14

i have also deleted the masternode.conf file to try and fix it but doesn't work
any ideas?

> 
2017-12-18 17:18:41 Dash Core version v0.12.2.1 (2017-11-11 10:02:45 +0300)
2017-12-18 17:18:41 InitParameterInteraction: parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1
2017-12-18 17:18:41 Error: Prune mode is incompatible with -txindex.
Error: Prune mode is incompatible with -txindex.
2017-12-18 17:18:41 PrepareShutdown: In progress...
2017-12-18 17:18:41 Verifying mncache.dat format...
2017-12-18 17:18:41 Loaded info from mncache.dat  1ms
2017-12-18 17:18:41      Masternodes: 0, peers who asked us for Masternode list: 0, peers we asked for Masternode list: 0, entries in Masternode list we asked for: 0, nDsqCount: 0
2017-12-18 17:18:41 Writing info to mncache.dat...
2017-12-18 17:18:41 Written info to mncache.dat  0ms
2017-12-18 17:18:41      Masternodes: 0, peers who asked us for Masternode list: 0, peers we asked for Masternode list: 0, entries in Masternode list we asked for: 0, nDsqCount: 0
2017-12-18 17:18:41 mncache.dat dump finished  1ms
2017-12-18 17:18:41 Verifying mnpayments.dat format...
2017-12-18 17:18:41 Loaded info from mnpayments.dat  1ms
2017-12-18 17:18:41      Votes: 0, Blocks: 0
2017-12-18 17:18:41 Writing info to mnpayments.dat...
2017-12-18 17:18:41 Written info to mnpayments.dat  0ms
2017-12-18 17:18:41      Votes: 0, Blocks: 0
2017-12-18 17:18:41 mnpayments.dat dump finished  3ms
2017-12-18 17:18:41 Verifying governance.dat format...
2017-12-18 17:18:41 Loaded info from governance.dat  0ms
2017-12-18 17:18:41      Governance Objects: 0 (Proposals: 0, Triggers: 0, Watchdogs: 0/0, Other: 0; Erased: 0), Votes: 0
2017-12-18 17:18:41 Writing info to governance.dat...
2017-12-18 17:18:41 Written info to governance.dat  0ms
2017-12-18 17:18:41      Governance Objects: 0 (Proposals: 0, Triggers: 0, Watchdogs: 0/0, Other: 0; Erased: 0), Votes: 0
2017-12-18 17:18:41 governance.dat dump finished  1ms
2017-12-18 17:18:41 Verifying netfulfilled.dat format...
2017-12-18 17:18:41 Loaded info from netfulfilled.dat  0ms
2017-12-18 17:18:41      Nodes with fulfilled requests: 0
2017-12-18 17:18:41 Writing info to netfulfilled.dat...
2017-12-18 17:18:41 Written info to netfulfilled.dat  0ms
2017-12-18 17:18:41      Nodes with fulfilled requests: 0
2017-12-18 17:18:41 netfulfilled.dat dump finished  1ms
root@OrangePeru-VM:/home/pool/coinbuilds/dash/src# 2017-12-18 17:18:41 Shutdown: done``

**Comment by PanderMusubi** [comment]
Fixing this would greatly improve the number of machines on which to run the software; contributing to the overall network speed.

**Comment by codablock** [comment]
I tested with -prune and -txindex=0, which gave me an error about prune not working with masternodes. This https://github.com/dashpay/dash/pull/1817 fixes this issue.

**Comment by nmarley** [comment]
This should be fixed with the `-litemode` argument once #2014 is merged in.

**Comment by PanderMusubi** [comment]
What is the current status? https://github.com/dashpay/dash/pull/2014 reports that it should be merged in 0.12.3. That version has been released, but is the PR included? If so, curious if users have been able to confirm that it has been solved or not.

**Comment by nmarley** [comment]
This was released in 0.12.3, with #2014 included.

It would be great if reporting users can confirm whether this is solved for them in the most recent release.

**Comment by PanderMusubi** [comment]
I can use `prune=555 -txindex=0` without an error but the wallet is not automatically starting to prune. What command can I run or what should I see in the debug log to see if prune is being activated or not?

**Comment by g1itch** [comment]
Seems still broken ):

**Comment by klenac** [comment]
I can confirm that it works (dashd-0.14.0.2). Just once, start dashd with the option -reindex-chainstate like this
```
dashd -prune=2300 -txindex=0 -reindex-chainstate
```
After some time (needed to sync and reindex) the blocks will be pruned and the space used will drop. Give it a few hours. You can monitor the CPU usage that is higher than normal during sync and reindexing, or you can check exactly how many blocks still remain to be processed using `./dash-cli getblockcount` and comparing the number with current block from dash blockchain explorer.

Next time start dashd without -reindex-chainstate option. You can add the options in dash.conf, so they will be picked up automatically. You must use both options, using prune alone generates the error in debug.log and exits.
```
prune=2300
txindex=0
```

**Comment by PastaPastaPasta** [comment]
When it says, "This mode is incompatible with -txindex" that means with txindex=1 or being enabled, setting txindex=0 specifically disables it.  I would say that this issue has been resolved, as shown by https://github.com/dashpay/dash/issues/1466#issuecomment-520542757.

**Comment by PastaPastaPasta** [comment]
This should be fixed. Please open a new issue if you're having problems with -prune
